packages/bitcore-node/src/models/webhook.ts

```diff
@@ -0,0 +1,54 @@
+import { ObjectID } from 'mongodb';
+import { StorageService } from '../services/storage';
+import { BaseModel } from './base';
+
+export interface IWebhook {
+  _id?: ObjectID;
+  chain: string;
+  network: string;
+  source: string;
+  sourceId: string;
+  tag?: string;
+  body: any;
+  timestamp: Date;
+  processed: boolean;
+}
+
+export class WebhookModel extends BaseModel<IWebhook> {
+  constructor(storage?: StorageService) {
+    super('webhook', storage);
+  }
+  allowedPaging = [];
+
+  onConnect() {
+    // capped at 0.5 million docs
+    this.db?.createCollection(this.collectionName, { capped: true, size: 500000 })
+      .catch((err) => { if (err.codeName !== 'NamespaceExists') throw err; });
+    this.collection.createIndex({ chain: 1, network: 1, source: 1 });
```

Comment:
suggestion: run in background to prevent collection locking `this.collection.createIndex({ chain: 1, network: 1, source: 1 }, { background: true });`